tool/src/demo_gen/terminus.rs

```diff
@@ -0,0 +1,390 @@
+                typescript: false,
+
+                imports: BTreeSet::new(),
+            },
+        };
+
+        // Not making this as part of the RenderTerminusContext because we want each evaluation to have a specific node,
+        // which I find easier easier to represent as a parameter to each function than something like an updating the current node in the struct.
+        let mut root =
+            MethodDependency::new(this.get_constructor_js(type_name.to_string(), method));
+
+        // And then we just treat the terminus as a regular constructor method:
+        this.terminus_info.node_call_stack = this.evaluate_constructor(method, &mut root);
+
+        this.terminus_info
+            .imports
+            .insert(
+                this.ctx
+                    .formatter
+                    .fmt_import_statement(&type_name, false, "../".into()),
+            );
+
+        Some(this.terminus_info)
+    }
+
+    /// Currently unused, plan to hopefully use this in the future for quickly grabbing parameter information.
+    fn _get_type_demo_attrs(&self, ty: &Type) -> Option<DemoInfo> {
+        match ty {
+            Type::Enum(e) => Some(e.resolve(self.ctx.tcx).attrs.demo_attrs.clone()),
+            Type::Opaque(o) => Some(o.resolve(self.ctx.tcx).attrs.demo_attrs.clone()),
+            Type::Struct(s) => Some(s.resolve(self.ctx.tcx).attrs.demo_attrs.clone()),
+            _ => None,
+        }
+    }
+
+    /// Take a parameter passed to a terminus (or a constructor), and either:
+    /// 1. Add it to the list of parameters that the terminus function takes for the render engine to call.
+    /// 2. Go a step deeper and look at its possible constructors to call evaluate_param on.
+    ///
+    /// `node` - Represents the current function of the parameter we're evaluating. See [`MethodDependency`] for more on its purpose.
+    fn evaluate_param(
+        &mut self,
+        param_type: &Type,
+        param_name: String,
+        node: &mut MethodDependency,
+        method_attrs: DemoInfo,
+    ) {
+        let attrs = Some(method_attrs); /*self.get_type_demo_attrs(param_type).or(Some(method_attrs));*/
+        // Helper function for quickly passing a parameter to both our node and the render terminus.
+        let out_param = |type_name| {
+            // This only works for enums, since otherwise we break the type into its component parts.
+            let label = attrs
+                .and_then(|attrs| {
+                    let label = attrs
+                        .input_cfg
+                        .get(&param_name)
+                        .map(|cfg| cfg.label.clone())
+                        .unwrap_or_default();
+
+                    if label.is_empty() {
+                        None
+                    } else {
+                        Some(label)
+                    }
+                })
+                .unwrap_or(param_name.clone());
```

Comment:
Suggestion: `param_name.clone()` could instead use `heck` to produce a nicer titlecase label